src/browser.js

```diff
@@ -19,25 +19,35 @@ export default function reactExpressMiddlewareGenerator (options = {}) {
 
 	return function reactExpressMiddleware (req, res, next) {
 		res.renderReactComponent = function renderReactComponent (Component, store, done) {
-			// store defaults to res.locals
-			if (typeof store === 'function') {
-				done = store;
-				store = undefined;
-			}
-			if (typeof store !== 'object') {
-				store = res.locals;
+			//
+			// Option 1: Store passed in, Component is a function
+			//
+			if (arguments.length > 1) {
+				// store defaults to res.locals
+				if (typeof store === 'function') {
+					done = store;
+					store = undefined;
+				}
+				if (typeof store !== 'object') {
+					store = res.locals;
 
-				// Override prototype on res.locals because react
-				// uses res.locals.hasOwnProperty and Express specefically
-				// uses Object.create(null) which means it doesn't have hasOwnProperty
-				setPrototypeOf(store, {});
-			}
+					// Override prototype on res.locals because react
+					// uses res.locals.hasOwnProperty and Express specefically
+					// uses Object.create(null) which means it doesn't have hasOwnProperty
+					setPrototypeOf(store, {});
+				}
 
-			// Create factory for component
-			Component = React.createFactory(Component);
+				// Create factory for component
+				Component = React.createFactory(Component);
```

Comment:
This can be consolidated so you dont have two lines calling `ReactDOM.render`, something like:

```
if (option1) {
  // ...
  Component = React.createFactory(Component)(store);
}
ReactDOM.render(Component, options.element, done);
```